tensorflow_probability/python/distributions/continuous_bernoulli.py

```diff
@@ -0,0 +1,454 @@
+            ),
+            (
+                tf.math.log(uniform * (2.0 * cut_lams - 1.0) + 1.0 - cut_lams)
+                - tf.math.log(1.0 - cut_lams)
+            )
+            / (tf.math.log(cut_lams) - tf.math.log(1.0 - cut_lams)),
+            uniform,
+        )
+        return tf.cast(sample, self.dtype)
+
+    def _cont_bern_log_norm(self):
+        lams = self._lams_parameter_no_checks()
+        cut_lams = tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            lams,
+            self._lims[0] * tf.ones_like(lams),
+        )
+        log_norm = tf.math.log(
+            tf.math.abs(2.0 * tf.math.atanh(1 - 2.0 * cut_lams))
+        ) - tf.math.log(tf.math.abs(1 - 2.0 * cut_lams))
+        taylor = (
+            tf.math.log(2.0)
+            + 4.0 / 3.0 * tf.math.pow(lams - 0.5, 2)
+            + 104.0 / 45.0 * tf.math.pow(lams - 0.5, 4)
+        )
+        return tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            log_norm,
+            taylor,
+        )
+
+    def _log_prob(self, event):
+        log_lams0, log_lams1 = self._outcome_log_lams()
+        event = tf.cast(event, log_lams0.dtype)
+        tentative_log_pdf = (
+            tf.math.multiply_no_nan(log_lams0, 1.0 - event)
+            + tf.math.multiply_no_nan(log_lams1, event)
+            + self._cont_bern_log_norm()
+        )
+        return tf.where(
+            tf.logical_or(event < 0, event > 1),
+            -float("Inf") * tf.ones_like(tentative_log_pdf),
+            tentative_log_pdf,
+        )
+
+    def _log_cdf(self, x):
+        return tf.math.log(self._cdf(x))
+
+    def _cdf(self, x):
+        lams = self._lams_parameter_no_checks()
+        cut_lams = tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            lams,
+            self._lims[0] * tf.ones_like(lams),
+        )
+        cdfs = (
+            tf.math.pow(cut_lams, x) * tf.math.pow(1.0 - cut_lams, 1.0 - x)
+            + cut_lams
+            - 1.0
+        ) / (2.0 * cut_lams - 1.0)
+        unbounded_cdfs = tf.where(
+            tf.logical_or(
+                tf.less(lams, self._lims[0]), tf.greater(lams, self._lims[1])
+            ),
+            cdfs,
+            x,
+        )
+        indicators0 = tf.where(
+            tf.logical_or(tf.less(x, 0.0), tf.greater(x, 1.0)),
+            tf.zeros_like(x),
+            tf.ones_like(x),
+        )
+        indicators1 = tf.where(
+            tf.greater(x, 1.0), tf.ones_like(x), tf.zeros_like(x)
+        )
+        return unbounded_cdfs * indicators0 + indicators1
+
+    def _log_survival_function(self, x):
+        return tf.math.log(self._survival_function(x))
+
+    def _survival_function(self, x):
```

Comment:
No need to implement this and the _log_survival_function. These use these implementations by default.